### PR TITLE
activate copilot inspection feature after dependent extensions are installed and activated and other miscellaneous features.

### PR DESCRIPTION
**Pull request by wangmingliang-ms**
1. [activate copilot inspection feature after dependent extensions are installed and activated.](https://github.com/microsoft/vscode-java-pack/commit/4202aaa8f5f67bce9179b3440d358b0a4d2b47ec) 
2. [support passing project context info(e.g. javaVersion) when sending request to copilot](https://github.com/microsoft/vscode-java-pack/commit/1b685c334cf04d3740c45db8794dea621499280d) 
3. [allow setting max concurrencies to prevent request spikes.](https://github.com/microsoft/vscode-java-pack/commit/051f15f17f79af239c0a272a83360ebd801895d8)

### CODE REVIEW

File: src/copilot/inspect/InspectionCopilot.ts [hunk 1]
```diff
@@ -132,43 +137,53 @@ export default class InspectionCopilot extends Copilot {
...
 
     public async inspectRange(document: TextDocument, range: Range | Selection): Promise<Inspection[]> {
-        // ajust the range to the minimal container class or (multiple) method symbols
-        const methods: SymbolNode[] = await getIntersectionMethodsOfRange(range, document);
-        const classes: SymbolNode[] = await getClassesContainedInRange(range, document);
-        const symbols: SymbolNode[] = [...classes, ...methods];
-        if (symbols.length < 1) {
-            const containingClass: SymbolNode = await getInnermostClassContainsRange(range, document);
-            symbols.push(containingClass);
+        if (this.busy) {
```
Comment (testforstephen) [inline src/copilot/inspect/InspectionCopilot.ts:140]: this.busy()?

File: src/copilot/inspect/index.ts [hunk 2]
```diff
@@ -30,4 +44,30 @@ async function rewrite(document: TextDocument, range: Range | Selection, _contex
...
+            return resolve();
+        }
+        const notInstalledExtensionIds = extensionIds.filter(id => !extensions.getExtension(id));
+        if (notInstalledExtensionIds.length > 0) {
+            sendInfo('java.copilot.inspection.dependentExtensions.notInstalledExtensions', { extensionIds: `[${notInstalledExtensionIds.join(',')}]` });
+            logger.info(`Dependent extensions [${notInstalledExtensionIds.join(', ')}] are not installed, setting checking interval to 10s.`);
+        } else {
+            logger.info(`All dependent extensions are installed, but some are not activated, keep checking interval ${interval}ms.`);
+        }
+        interval = notInstalledExtensionIds ? interval : 10000;
```
Comment (testforstephen) [inline src/copilot/inspect/index.ts:63]: According to your comment, if there are dependent extensions not installed, the checking interval is 10s.

`notInstalledExtensionIds.length ? 10000 : interval` ?

File: src/copilot/inspect/InspectionCopilot.ts [hunk 3]
```diff
@@ -132,43 +137,53 @@ export default class InspectionCopilot extends Copilot {
...
+            logger.warn('Copilot is busy, please retry after current inspecting tasks is finished.');
+            void window.showWarningMessage(`Copilot is busy, please retry after current inspecting tasks are finished.`);
```
Comment (testforstephen) [inline src/copilot/inspect/InspectionCopilot.ts:142]: When I open more than three Java files, the concurrency exceeds the limit of three. Do you think it’s good to display a message even if the inspection is not manually triggered by the user?

**Review by testforstephen — CHANGES_REQUESTED** [review]

File: src/copilot/inspect/InspectionCopilot.ts (hunk 1 above)
Comment (wangmingliang-ms) [reply src/copilot/inspect/InspectionCopilot.ts:140]: `get busy()` is a getter, no need to add `()`

File: src/copilot/inspect/index.ts (hunk 2 above)
Comment (wangmingliang-ms) [reply src/copilot/inspect/index.ts]: fixed.

**Review by testforstephen — APPROVED** [review]

**Pull request merged**